src/result_processor.c

```diff
@@ -574,26 +641,33 @@ int loader_Next(ResultProcessorCtx *ctx, SearchResult *r) {
     return rc;
   }
   Document doc = {NULL};
-  RedisModuleKey *rkey = NULL;
 
   // Current behavior skips entire result if document does not exist.
   // I'm unusre if that's intentional or an oversight.
-  RedisModuleString *idstr = DMD_CreateKeyString(r->scorerPrivateData, lc->ctx->redisCtx);
+  const RSDocumentMetadata *dmd = DocTable_Get(&lc->ctx->spec->docs, r->docId);
+  if (dmd == NULL || (dmd->flags & Document_Deleted)) {
+    return RS_RESULT_OK;
```

Comment:
shouldn't we just skip the document if it was remove, and if not pls add a comment why ...

Reply:
I don't understand the question- this seems like what we're doing.